### PR TITLE
Improvised turret

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Content "Restore obsoleted 9mm turret as improvised turret"

#### Purpose of change
The 9x19mm turret was essentially obsoleted in #33436. This returns it to something used in the game. Recipes for the other two turrets were inappropriate since the turrets were changed to complex machinery.

#### Describe the solution
- Adjusted the `mon_turret` definition to make it an improvised turret.
- Removed recipes for non-improvised turrets.
- Added `bot_antimateriel` to robots itemgroup. Previously this did not spawn.

### CODE REVIEW

**Comment by Rail-Runner** [comment]
> Removed recipes for non-improvised turrets.

I think this is unnecessary. In #30456 a robot salvage system was added; all robots (including turrets) are built using proper components salvaged from other similar robots that themselves can't be crafted, so they aren't really even improvised; more like assembing salvaged factory-made modules into a finished product.

**Comment by ymber** [comment]
Restored salvage recipes for proper turrets and allowed improvising some components of improvised turret.

File: data/json/recipes/recipe_electronics.json [hunk 1]
```diff
@@ -1268,29 +1268,33 @@
...
-      [ [ "gun_module", 2 ] ],
-      [ [ "hk_mp5", 2 ] ],
+      [ [ "camera", 1 ] ],
       [ [ "small_storage_battery", 1 ] ],
-      [ [ "sensor_module", 1 ] ],
-      [ [ "identification_module", 1 ] ],
       [ [ "power_supply", 1 ] ],
-      [ [ "turret_chassis", 1 ] ]
+      [ [ "motor_tiny", 3 ] ],
+      [ [ "hk_mp5", 1 ] ],
```
Comment (Rail-Runner) [inline data/json/recipes/recipe_electronics.json]: I think recipe would need an MP5 magazine as well, to hold the ammo.

**Comment by LaVeyanFiend** [comment]
If it's an _improvised_ turret, does it still have to require specifically an MP5? There's more than 1 automatic 9mm gun ingame.

**Comment by Rail-Runner** [comment]
> If it's an _improvised_ turret, does it still have to require specifically an MP5? There's more than 1 auto-capable 9mm gun ingame.

Probably not, but I think there's currently no way to allow installing an arbitrary gun onto a turret. If you just add any other gun alternatives to recipe, then gun stats won't match.

File: data/json/monsters/turrets.json [hunk 2]
```diff
@@ -20,10 +20,26 @@
...
     "special_attacks": [
-      { "type": "gun", "cooldown": 1, "gun_type": "hk_mp5", "ranges": [ [ 0, 6, "AUTO" ], [ 7, 14, "DEFAULT" ] ] },
-      [ "PARROT", 40 ]
+      {
+        "type": "gun",
+        "cooldown": 1,
+        "move_cost": 150,
+        "gun_type": "hk_mp5",
+        "ammo_type": "9mm",
+        "fake_skills": [ [ "gun", 5 ], [ "rifle", 5 ] ],
```
Comment (Rail-Runner) [inline data/json/monsters/turrets.json]: May want to replace rifle skill with SMG skill here, since that's the skill needed by MP5.

**Comment by tenmillimaster** [comment]
@Rail-Runner could NPC weapon handling code be co-opted for the turret? E.g. deploying the turret spawns a weaponless monster, and you hand it a compatible, loaded gun?

**Comment by Rail-Runner** [comment]
> could NPC weapon handling code be co-opted for the turret? E.g. deploying the turret spawns a weaponless monster, and you hand it a compatible, loaded gun?

I don't know that.

**Pull request merged**